src/riscv.c

```diff
@@ -105,11 +106,65 @@ riscv_word_t rv_get_reg(riscv_t *rv, uint32_t reg)
     return ~0U;
 }
 
-riscv_t *rv_create(const riscv_io_t *io,
-                   riscv_user_t userdata,
-                   int argc,
-                   char **args,
-                   bool output_exit_code)
+/*
+ * Remap stdio
+ *
+ * @rv: riscv
+ * @fsp: a list of pair of mapping from fd to FILE *
+ * @fsp_size: list size
+ *
+ * Note: fd inside fsp should be 0 or 1 or 2 only
+ *
+ */
+void rv_remap_stdstream(riscv_t *rv, fd_stream_pair_t *fsp, uint32_t fsp_size)
+{
+    if (!rv)
+        return;
+
+    vm_attr_t *attr = PRIV(rv);
+    if (!attr)
+        return;
+    if (!attr->fd_map)
+        return;
+
+    for (uint32_t i = 0; i < fsp_size; i++) {
+        int fd = fsp[i].fd;
+        FILE *file = fsp[i].file;
+        if (fd != 0 && fd != 1 && fd != 2)
```

Comment:
Maybe it's not a great idea to use 0, 1, and 2 here directly. @jserv Do you think we should define macros for each and expand them to the correct value considering the operating system(e.g.  `STDIN_FILENO` for POSIX and _fileno(stdin) on WIN)?

Reply:
See [commit](https://github.com/sysprog21/rv32emu/pull/340/commits/f953ac1e051558623ae3d1bfd4cee70077e47c53#diff-f99b8d1f48ef2257a9bbbe6b487c621f74ef233c00858df5a37b06e89441e836)